game/tap.ksy

```diff
@@ -0,0 +1,49 @@
+meta:
+  id: tap
+  file-extension: tap
+  endian: le
+  license: CC0-1.0
+  title: ZX Spectrum tape file format
+doc-ref: https://faqwiki.zxnet.co.uk/wiki/TAP_format
+seq:
+  - id: block
+    type: block
+    repeat: eos
+enums:
+  flag_enum:
+    0x00: header
+    0xFF: data
+  header_type_enum:
+    0: program
+    1: num_array
+    2: char_array
+    3: code_file
+types:
+  block:
+    seq:
+      - id: length
+        contents: [0x13, 0x00]
+      - id: flag
+        type: u1
+        enum: flag_enum
+      - id: header
+        type: header_block
+      - id: data
+        size: header.data_length + 4
+  header_block:
+    seq:
+      - id: header_type
+        type: u1
+        enum: header_type_enum
+      - id: filename
+        size: 10
+        type: str
+        encoding: ASCII
```

Comment:
Probably this warrants certain extension to character sets that will automatically decode that into Unicode where possible, but this is definitely a separate big feat. If wonder if any big organization have already assigned ZX Spectrum encoding some kind of codepage/official name, and if that has support in something like iconv?

Please add this as a reminder to https://github.com/kaitai-io/kaitai_struct/issues/393 ?

Reply:
This may be of interest: [L2/19-025: Proposal to add characters from legacy computers and teletext to the UCS](https://www.unicode.org/L2/L2019/19025-terminals-prop.pdf).

One thing to keep in mind is that the ZX Spectrum character set includes user-defined graphics characters (basically characters whose glyphs you can change as you like) and BASIC tokens (`PRINT`, `LOAD`, `IF`, etc. - these are actually single characters, but are normally displayed as multiple characters on screen). There's no obvious answer to transcoding these - especially with the BASIC keywords it depends on what you want to do with the text. For displaying you want to "expand" them to their text equivalents, but when comparing or storing them you want to keep them in their original byte form.